visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts

```diff
@@ -41,12 +41,29 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		ThreeOrbitControlsService.subscribe(this.$rootScope, this)
 	}
 
-	//labels need to be scaled according to map or it will clip + looks bad
-	addLabel(node: Node, options: { showNodeName: boolean; showNodeMetric: boolean }, highestNodeInSet: number) {
-		const state = this.storeService.getState()
+	// Labels need to be scaled according to map or it will clip + looks bad
+	addLabel(node: Node, highestNodeInSet: number, enforceLabel = false) {
+		const { appSettings, dynamicSettings, treeMap } = this.storeService.getState()
 
-		const { scaling, layoutAlgorithm } = state.appSettings
-		const { margin, heightMetric } = state.dynamicSettings
+		const { scaling, layoutAlgorithm, showMetricLabelNodeName, showMetricLabelNameValue } = appSettings
+		const { margin, heightMetric } = dynamicSettings
+
+		let labelText = ""
+
+		if (showMetricLabelNodeName || (enforceLabel && !showMetricLabelNameValue)) {
+			labelText = `${node.name}`
+		} else if (!showMetricLabelNameValue) {
+			return
+		}
+
+		if (showMetricLabelNameValue) {
+			if (labelText !== "") {
+				labelText += "\n"
+			}
+			labelText += `${node.attributes[heightMetric]} ${heightMetric}`
```

Comment:
Wouldn't be the color metric more appropriate?

Reply:
That would change the behavior from the current one. I personally would keep it as is for now.